docs/stopping_events/syscalls.md

```diff
@@ -0,0 +1,206 @@
+| `recursive` | `bool` | If set to `True`, the handler's callback will be executed even if the syscall was triggered by a hijack. |
+
+**Returns**:
+
+| Return | Type | Description |
+| --- | --- | --- |
+| `SyscallHandler` | [SyscallHandler](../../from_pydoc/generated/data/syscall_handler) | The handler object created. |
+
+### Callback Signature
+
+!!! ABSTRACT "Callback Signature"
+    ```python
+    def callback(t: ThreadContext, handler: HandledSyscall) -> None:
+    ```
+
+**Parameters**:
+
+| Argument | Type | Description |
+| --- | --- | --- |
+| `t` | [ThreadContext](../../from_pydoc/generated/state/thread_context) | The thread that hit the syscall. |
+| `handler` | [SyscallHandler](../../from_pydoc/generated/data/syscall_handler) | The SyscallHandler object that triggered the callback. |
+
+!!! INFO "Nuances of Syscall Handling"
+    The syscall handler is the only [stopping event](../stopping_events/) that can be triggered by the same syscall twice in a row. This is because the handler is triggered both when the syscall is entered and when it is exited. As a result the `hit_on()` method of the [SyscallHandler](../../from_pydoc/generated/data/syscall_handler) object will return `True` in both instances.
+
+    You can also use the `hit_on_enter()` and `hit_on_exit()` functions to check if the cause of the process stop was the syscall entering or exiting, respectively.
+
+    As for the `hit_count` attribute, it only stores the number of times the syscall was *exited*.
+
+---
+
+!!! ABSTRACT "Example usage of asynchronous syscall handlers"
+    ```python
+    def on_enter_open(t: ThreadContext, handler: SyscallHandler):
+        print("entering open")
+        t.syscall_arg0 = 0x1
+
+    def on_exit_open(t: ThreadContext, handler: SyscallHandler):
+        print("exiting open")
+        t.syscall_return = 0x0
+
+    handler = d.handle_syscall(syscall="open", on_enter=on_enter_open, on_exit=on_exit_open)
+    ```
+
+!!! ABSTRACT "Example of synchronous syscall handling"
+    ```python
+    from libdebug import debugger
+
+    d = debugger("./test_program")
+    d.run()
+
+    handler = d.handle_syscall(syscall="open")
+    d.cont()
+
+    if handler.hit_on_enter(d):
+        print("open syscall was entered")
+    elif handler.hit_on_exit(d):
+        print("open syscall was exited")
+    ```
+
+    The script above will print "open syscall was entered".
+
+## :octicons-number-24: Resolution of Syscall Numbers
+Syscall handlers can be created with the identifier number of the syscall or by the syscall's common name. In the second case, syscall names are resolved from a definition list for Linux syscalls on the target architecture. The list is fetched from [mebeim's syscall table](https://syscalls.mebeim.net). We thank him for hosting such a precious resource. Once downloaded, the list cached internally. 
```

Comment:
Once downloaded, the list *gets* cached internally.

Reply:
Addressed in [288cb8b](https://github.com/libdebug/libdebug/commit/288cb8b5bd711382cc6116993dfb4c94cf498426)